### DISCUSSION TITLE
Discuss automatic technology identification based template execution

### DISCUSSION THREAD

**Discussion by Ice3man543** [projectdiscovery/nuclei#1477] [Ideas]
Basically, a form of smart workflows, where first the technology is detected, and only the identified templates for technologies are executed. We need to discuss how we'd like to accomplish this, creating this issue for tracking purposes.

**Comment by pdelteil** [comment]
Hello,

This would be very interesting and time saver. 

I think the first problem are detection rules, some technologies require different type of requests or technique to find them all (e.g. ColdFusion, Confluence, etc). It would be problematic for me if the detection file/template is to difficult to understand/modify or debug.
